core/modfunc.go

```diff
@@ -100,6 +101,22 @@ func (fn *ModuleFunction) Call(ctx context.Context, caller *idproto.ID, opts *Ca
 	}
 	ctx = bklog.WithLogger(ctx, lg)
 
+	analytics.Ctx(ctx).Capture(
+		"module_call",
+		map[string]any{
+			"function_name": fn.metadata.Name,
+
+			"module_name":    mod.Name(),
+			"module_id:":     "FIXME",
+			"module_ref":     "FIXME", // FIXME: this should be inferred by module_id
+			"module_version": "FIXME", // FIXME: this should be inferred by module_id
+
+			"caller_id":      "FIXME",
+			"caller_ref":     "FIXME", // FIXME: this should be inferred by caller_id
+			"caller_version": "FIXME", // FIXME: this should be inferred by caller_id
```

Comment:
@aluzzardi FYI https://github.com/dagger/dagger/pull/6386 is merged, so once this is rebased then `module_*` fields should be accessible via e.g. `fn.mod.Source.Self`.

I *believe* the `caller_*` fields will be obtainable via `fn.mod.Query.CurrentModule()`, though need to be sure to handle error there, which is returned when the call is coming from not a module (i.e. a caller on the host like the CLI).
* Might make sense to tweak `fn.mod.Query.CurrentModule()` to not error out in that case and instead return `(dagql.Id[*Module, bool, error)`, with the extra bool indicating that the caller is not from a module.
* This does rely on the plumbing forwarding all the grpc headers that we use to track module callers, which I *think* is the case, but worst case there may just need to be a few other tweaks to plumbing; the information is fundamentally accessible now.